File: avalon-extensions/avalon-pokeapi/avalon-pokeapi-exporter/src/test/java/io/github/lishangbu/avalon/pokeapi/TestEnvironmentApplication.java
package io.github.lishangbu.avalon.pokeapi;

import io.github.lishangbu.avalon.json.autoconfiguration.JacksonAutoConfiguration;
import io.github.lishangbu.avalon.pokeapi.autoconfiguration.PokeDataComponentAutoConfiguration;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Import;

/// 测试环境自动配置类
///
/// @author lishangbu
/// @since 2025/8/20
@Import(
        value = {
            PokeApiDataProviderAutoConfiguration.class,
            PokeDataComponentAutoConfiguration.class,
            JacksonAutoConfiguration.class
        })
@Configuration(proxyBeanMethods = false)
public class TestEnvironmentApplication {}

File: avalon-modules/avalon-dataset/avalon-dataset-repository/src/test/kotlin/io/github/lishangbu/avalon/dataset/TestEnvironmentApplication.kt
package io.github.lishangbu.avalon.dataset

import org.springframework.boot.SpringBootConfiguration
import org.springframework.boot.autoconfigure.EnableAutoConfiguration
import org.springframework.boot.testcontainers.service.connection.ServiceConnection
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.ComponentScan
import org.testcontainers.postgresql.PostgreSQLContainer
import org.testcontainers.utility.DockerImageName

/**
 * 数据集模块测试环境配置
 *
 * 为仓储测试提供最小化的 Spring Boot 上下文
 */
@SpringBootConfiguration
@EnableAutoConfiguration
@ComponentScan(basePackages = ["io.github.lishangbu.avalon.dataset.repository"])
class TestEnvironmentApplication {
    /** 创建 PostgreSQL 测试容器 */
    @Bean
    @ServiceConnection
    fun postgresContainer(): PostgreSQLContainer = PostgreSQLContainer(DockerImageName.parse("postgres"))
}
